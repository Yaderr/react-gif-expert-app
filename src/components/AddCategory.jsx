import { useState } from "react"
import propTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')
    
    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSearch = (e) => {
        e.preventDefault()
        if(inputValue.trim().length <= 1) return 
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return (
        <>
            <div className="addCategory-container">
                <form onSubmit={ onSearch } aria-label="form">
                    <input 
                        value={ inputValue }
                        onChange={ onInputChange }
                        type="search" 
                        name="" 
                        placeholder="Busca todos lo gif" 
                    />
                </form>
            </div>
        </>
    )
}

AddCategory.propTypes = {
    onNewCategory: propTypes.func.isRequired
}