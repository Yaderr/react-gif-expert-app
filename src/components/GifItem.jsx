
export const GifItem = ({ gif }) => {
    return (
        <>
           <div className="gif-card">
                <img src={gif.url} alt="" width="100%" />
                <div className="gif-card-title">
                    <h6>{gif.title}</h6>
                </div>
           </div>
        </>
    )
}