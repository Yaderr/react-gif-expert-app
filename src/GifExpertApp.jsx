import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'Noticias', 'Tractor' ])

    const addNewCategory = (newCategory) => {
        if(categories.includes(newCategory)) return
        setCategories([newCategory, ...categories])
    }
    
    return (
        <>
            <h1>Gif Expert App</h1>
            <AddCategory onNewCategory={ addNewCategory } />
            {
                categories.map(category => (
                    <GifGrid key={ category } category={ category } />
                ))
            }            
        </>
    )
}