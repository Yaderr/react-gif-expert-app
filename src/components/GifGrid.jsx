import { GifItem } from "./GifItem"
import { useFetchGif } from "../hooks/useFetchGif"

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGif(category)
    
    return (
        <>
            <div className="content">
                <h2>{ category }</h2>
                {
                    isLoading 
                    ? (<h2>Loading...</h2>)
                    : null
                }
                <div className="gif-grid">
                    {images.map(img => (
                        <GifItem key={img.id} gif={img} />
                    ))}
                </div>
            </div>
        </>
    )
}