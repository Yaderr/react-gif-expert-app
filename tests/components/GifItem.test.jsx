import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('GifItem component tests', () => {
    const testGif = {
        url: 'https://media1.giphy.com/media/Y82PJo20tfF1n48vPR/giphy.gif',
        title: 'My Test Gif'
    }

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem gif={testGif} />)
        expect(container).toMatchSnapshot()
    })

    test('debe tener un h6 con el titulo', () => {
        render(<GifItem gif={testGif} />)
        expect(screen.getByRole('heading', { level: 6 }).innerHTML).toBe(testGif.title)
    })

    test('debe tener una imagen con la url en el src', () => {
        render(<GifItem gif={testGif} />)
        expect(screen.getByRole('img').src).toBe(testGif.url)
    })
})