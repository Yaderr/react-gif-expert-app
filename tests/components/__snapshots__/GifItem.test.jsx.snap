// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem component tests debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="gif-card"
  >
    <img
      alt=""
      src="https://media1.giphy.com/media/Y82PJo20tfF1n48vPR/giphy.gif"
      width="100%"
    />
    <div
      class="gif-card-title"
    >
      <h6>
        My Test Gif
      </h6>
    </div>
  </div>
</div>
`;
